src/main/java/com/hpe/adm/octane/ideplugins/services/mywork/DynamoMyWorkService.java

```diff
@@ -218,6 +219,32 @@ public boolean removeFromMyWork(EntityModel entityModel) {
         return false;
     }
 
+    @Override
+    public EntityModel getEntityFromUserItemIfNeeded(EntityModel entity) {
+        if(Entity.USER_ITEM != Entity.getEntityType(entity)){
+            throw new ServiceRuntimeException("Given param entity is not of type: user_item, type is: " + Entity.getEntityType(entity));
+        }
+        String followField = "my_follow_items_" + entity.getValue("entity_type").getValue();
```

Comment:
can this throw a NPE?

Reply:
Teoretically, no. The entities will always have type. It never threw an NPE. I can add a checker if needed :)

Reply:
W: I see there's a fair amount of clean code here, so why don't you continue doing clean code urself 
1) Extract the check if u require entity from UserItem in a different method (which can be reused in all cases - with stream().filter() and standard if case where applicable) 
2) Extract the rest of the method in "getEntityFromUserItem" (ifNeeded seems like doing 2 things into a single one)

Results
-> clean code 
-> better readability
-> no more methods doing X things instead one single good thing. 


I marked it as "would", but it's totally a good addition to the code.

Reply:
> Teoretically, no. The entities will always have type. It never threw an NPE. I can add a checker if needed :)

is there any case where users cand do REST calls with this? If yes, you really need to check NPE

Reply:
M: I won't comment for all classes. But some refactor is in place - you have a lot of duplicated code for literally the same thing. Extract it and reuse it as required.